Llm integration bug fix

jnduli opened the pull request:
# Description

Bug fixes for LLM integration work that includes:

- ensure `gnqa_answers` jinja templates extends `base.html`
- send `get` request instead of `put` when accessing gn3's llm search api
- provide a rough implementation for the `xapian_query_cleaner` implementation.
- add a spinner when api search is happening

# Tested

Results after search screenshot:
![image](https://github.com/user-attachments/assets/26d9befb-ca13-4b95-8c0c-033f97095288)

And the ai results page:
![image](https://github.com/user-attachments/assets/35b028af-bff9-4ea8-b04f-96190fedb399)

jnduli committed:
fix: gnqa_answer but fix for missing html

jnduli committed:
fix: change request for search to get

jnduli committed:
feat: add rough implementation for xapian query cleaner

Alexanderlacuna reviewed (changes requested):
"I don’t think you should extend `base.html`. The `gnqa_answer` page is a subpage of `gnqa` that  extends `base.html`, which is rendered dynamically using htmx."

Alexanderlacuna commented:
Related to the above(context screenshot)
Question : Wasn't "See reference" supposed to be a button that opens to a  new webpage in this case the  gnqa_anwer page??

Alexanderlacuna commented on gn2/wqflask/views.py:
```diff
@@ -270,9 +270,21 @@ def gsearchtable():
+        if len(split_word) > 0 and split_word[0].lower() in xapian_prefixes:
+            final_query.append(split_word[1])
+            continue
+        if split_word[0].lower() in range_prefixes:
```
this  looks good

jnduli committed:
fix: stop extending gnqa_answer.html

gnqa_answer.html is used by htmx to return a post result. I revert my
change and include this in `gnqa.html` instead, to load results from a
GET request.

jnduli commented:
> "I don’t think you should extend `base.html`. The `gnqa_answer` page is a subpage of `gnqa` that extends `base.html`, which is rendered dynamically using htmx."

Thanks for catching this. I've fixed this and included `gnqa_answer` in the `gnqa.html` page when we send a GET request.

jnduli commented:
> Related to the above(context screenshot) Question : Wasn't "See reference" supposed to be a button that opens to a new webpage in this case the gnqa_anwer page??

Yeah, you're right. However when I used the `gnqa_answer.html` pages, it didn't look good since it didn't extend base.html. Right now the change I've done uses `gnqa.html` and loads up the content for this.

The `References` link still opens a new tab, but in this case it opens up the `GNQA` search page with results already pre-populated.

jnduli commented:
Here's how the search page looks like with GET requests:
![image](https://github.com/user-attachments/assets/28b175f4-1b17-48cc-8759-cca54c637c1d)
